Feature/#174058751 update set gamerosters frontend to new bulk upload backend

marclanepitt opened the pull request.

hmgoalie35 reviewed (changes requested):
Looks good, a few things I noticed when QAing

* The bulk endpoints only allow 50 items at a time. It's not likely but it is possible for a bulk create or bulk delete call to contain > 50 items (if there are 25+ players for both teams). I don't think it is worth the effort to have the frontend batch api calls so i was thinking
  * Update the api to accept 75 items at a time. 50 was chosen because the api wasn't too slow when processing that many items but maybe 75 works also. If we go this route probably want to make sure the api isn't too slow
  * We could alternatively show an error msg to the user saying they can only choose x items at a time, but that might be confusing. The api returns a 400 with an error msg regarding only x items at a time so might want to just display that error instead of the `toggleAPIError` function. The frontend needs to handle 4xx errors better in general, it's possible for a game's team to change when someone is editing the rosters which would result in 4xx errors.
* Can you change the PR title to match the conventions so pivotal tracker picks it up
* I noticed some weird behavior, not sure if it is expected. Follow these steps
  * Add a player to the roster, then press update
  * Remove player from roster, do not press updated
  * Add player back, the frontend does an update api call
  * Repeat the above steps but w/ goalie players and setting the goalies to starters
  * Actually maybe this is expected behavior, I need to check again

hmgoalie35 commented:
I double checked and if I
* Add a player that is a goalie, set them as starter & save
* Remove them and re-add them (all without pressing save)
* The fact that they were marked as a starter is not preserved (the ui shows the `Set as Starter` text)
* Pressing update doesn't actually set is_starting to false in the db, you'll actually see it switch to `Starter` after the api call completes

marclanepitt commented:
> I double checked and if I
> 
> * Add a player that is a goalie, set them as starter & save
> * Remove them and re-add them (all without pressing save)
> * The fact that they were marked as a starter is not preserved (the ui shows the `Set as Starter` text)
> * Pressing update doesn't actually set is_starting to false in the db, you'll actually see it switch to `Starter` after the api call completes

Yep definitely not expected behavior, went ahead and pushed a fix for this

marclanepitt commented:
> Looks good, a few things I noticed when QAing
> 
> * The bulk endpoints only allow 50 items at a time. It's not likely but it is possible for a bulk create or bulk delete call to contain > 50 items (if there are 25+ players for both teams). I don't think it is worth the effort to have the frontend batch api calls so i was thinking
>   
>   * Update the api to accept 75 items at a time. 50 was chosen because the api wasn't too slow when processing that many items but maybe 75 works also. If we go this route probably want to make sure the api isn't too slow
>   * We could alternatively show an error msg to the user saying they can only choose x items at a time, but that might be confusing. The api returns a 400 with an error msg regarding only x items at a time so might want to just display that error instead of the `toggleAPIError` function. The frontend needs to handle 4xx errors better in general, it's possible for a game's team to change when someone is editing the rosters which would result in 4xx errors.
> * Can you change the PR title to match the conventions so pivotal tracker picks it up
> * I noticed some weird behavior, not sure if it is expected. Follow these steps
>   
>   * Add a player to the roster, then press update
>   * Remove player from roster, do not press updated
>   * Add player back, the frontend does an update api call
>   * Repeat the above steps but w/ goalie players and setting the goalies to starters
>   * Actually maybe this is expected behavior, I need to check again

I added some functionality to display custom messages on the untimed error banner. I chose only to display only non_field errors, as they seem the most useful to the user. Any field error is controlled by us and displaying the error message wouldn't be helpful.

Agreed that we should up the bulk item max

Updated title

Update bug you pointed out

hmgoalie35 commented on static/js/gameRosters/GameRostersUpdateComponent.js:
```diff
@@ -91,81 +127,255 @@ export default class GameRostersUpdateComponent extends React.Component {
+            createNotification('Some of your updates were not saved. Please refresh and try again.', 'warning').show();
+          }
+        } else {
+          let message = `We experienced ${errors.length} issue${errors.length > 1 ? 's' : ''} with your request`;
```
there is a `pluralize` util location in `utils/index.js` we could use

hmgoalie35 replied:
I pushed a fix for this

hmgoalie35 reviewed (commented):
Pushed the fix using the pluralize util, tests seem to be failing

hmgoalie35 commented:
Noticed 2 other things

* If you try creating > 50 items (resulting in a 400 api response) the user is left kind of confused because the error message isn't visible until they scroll up. Maybe we should auto scroll to the top?
* If you try creating > 50 items, update, then fix things so there are < 50 items and update the error msg banner is still present
